api/v1alpha1/grant_types.go

```diff
@@ -83,6 +85,13 @@ func (g *Grant) MariaDBRef() *MariaDBRef {
 	return &g.Spec.MariaDBRef
 }
 
+func (g *Grant) HostnameOrDefault() string {
```

Comment:
What about adding the `AccountName` method to the `Grant` for the sake of unification?

Reply:
@mmontes11 Sure could do that but will also imply that we change the `GrantOpts` struct definition, replacing the `Username` and `Host` field by a `AccountName` string.

Also it seems like the `Privileges`, `Database`, `Table` and `AccountName` fields of the `GrantOpts` are always required. What do you think about changing the signature of the `Grant` and `Revoke` methods so they take these values as regular parameters and we would only keep the optional arguments of the option in the `GrantOpts` 🤔

Also does not seem like the [`REVOKE` statement](https://mariadb.com/kb/en/revoke/) takes any optional argument so what about dropping the `GrantOpts` from the signature? And if needed we later we could add a dedicated `RevokeOpts` struct 😄 

Something such as:
```go
func (c *Client) Grant(ctx context.Context, privileges []string, database, table, accountName string, opts GrantOpts) error

func (c *Client) Revoke(ctx context.Context, privileges []string, database, table, accountName string) error
```

Reply:
In that case the `GrantOpts` would be just the `GrantOption` right?
```golang
type GrantOpts struct {
	GrantOption bool
}
```
We should still support revoking the `GRANT OPTION` so we would need to pass the `GrantOpts` to `Revoke` as well. For indicating that is nullable, we could pass it with a pointer, as follows:
```golang
func (c *Client) Grant(ctx context.Context, privileges []string, database, table, accountName string, opts *GrantOpts) error

func (c *Client) Revoke(ctx context.Context, privileges []string, database, table, accountName string,  opts *GrantOpts) error
```
What do you think?

Reply:
Yes only the `GrantOption` will remain. Didn't take a closer look at the `GRANT` statement document but there's might be more support options by MariaDB itself though.

Passing a pointer to suggest the fact that the `GrantOpts` is optional sounds good. We could  also pass it as a [functional option](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) so other developers who don't need options when calling these methods doesn't even need to pass any value when calling it.

I'll push an implementation example to make it easier to visualize 😉

Reply:
Agreed, let's go with an options pattern interface. You have an example here:
- https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/controller/galera/controller.go#L53